spec.bs

```diff
@@ -4322,6 +4368,30 @@ navigating to another page. Some implementations, such as Chromium, have chosen
         Issue:  Serializing an Infra value to JSON bytes expects to be called within a valid ES realm. See
         <a href="https://github.com/whatwg/infra/issues/625">infra/625</a>
 
+        <dt>"`adSizes`"
+        <dd>
+        1. Let |adSizes| be a new [=map=] whose [=map/keys=] are [=strings=] and [=map/values=] are
+           [=ad sizes=].
+        1. [=map/For each=] |sizeName| → |size| of |value|:
+          1. If |sizeName| is "", [=exception/throw=] a {{TypeError}}.
+          1. Let |adSize| be the result from running [=parse an AdRender ad size=] with |size|.
+          1. If |adSize| is null, [=exception/throw=] a {{TypeError}}.

```

Comment:
This algorithm [runs in parallel](https://pr-preview.s3.amazonaws.com/gtanzer/turtledove/pull/1210.html#ref-for-interest-group-update%E2%91%A2) so you cannot throw errors from it.

Reply:
Oops, I noticed the whole "jump to the step labeled <i><a href=#abort-update>Abort update</a></i>." thing but forgot to use it.